pkg/server/vault.go

```diff
@@ -0,0 +1,405 @@
+	if err != nil {
+		return fmt.Errorf("failed to enable KV engine at path %s: %v", path, err)
+	}
+
+	if resp.StatusCode != 204 {
+		return fmt.Errorf("failed to enable KV engine, unexpected status code: %v", resp.StatusCode)
+	}
+
+	fmt.Printf("Successfully enabled KV version 2 at path: %s\n", path)
+	return nil
+}
+
+// StartClient starts a local Vault server in development mode.
+//
+// This function is only called when using a locally installed Vault. It performs the following steps:
+// 1. Kills any existing Vault process.
+// 2. Starts a new Vault server in development mode with a predefined root token.
+// 3. Sets the VAULT_ADDR environment variable based on the operating system.
+// 4. Enables the KV secrets engine at the "secret" path.
+//
+// Returns:
+// - *VaultError: An error object if the Vault process fails to start or if setting the environment variable fails.
+func (v *VaultClient) StartClient() *VaultError {
+	// Kill existing Vault process
+	v.CloseClient()
+	// Start Vault in dev mode
+	vaultCmd = exec.Command("vault", "server", "-dev", "-dev-root-token-id="+v.client.Token())
+	vaultCmd.Stdout = os.Stdout
+	vaultCmd.Stderr = os.Stderr
+
+	// Set environment variable based on OS
+	if runtime.GOOS == "windows" {
+		// For Windows (cmd.exe and PowerShell)
+		err := os.Setenv("VAULT_ADDR", "http://127.0.0.1:8200")
+		if err != nil {
+			return &VaultError{
+				Message: fmt.Sprintf("failed to set VAULT_ADDR: %v", err),
+			}
+		}
+	} else {
+		// For Linux/macOS
+		os.Setenv("VAULT_ADDR", "http://127.0.0.1:8200")
+	}
+
+	if err := vaultCmd.Start(); err != nil {
+		return &VaultError{
+			Message: fmt.Sprintf(VaultErrorRunning, err),
+		}
+	}
+	time.Sleep(1 * time.Second)
+
+	_ = enableKVSecretsEngine(v.client, "secret")
+	//if sErr != nil {
+	//	return &VaultError{Message: fmt.Sprintf("%v", sErr)}
+	//}
+	return nil
+}
+
+// WriteSecret stores a key-value pair in the Vault secrets engine in development mode.
+//
+// This function is intended for use with a local Vault instance. It validates input parameters
+// and writes the specified key-value pair to the given path in Vault.
+//
+// Parameters:
+// - path: The Vault path where the secret should be stored (e.g., "secret/myapp").
+// - key: The name of the key to store in the secret (e.g., "API_KEY").
+// - value: The value associated with the key.
+//
+// Returns:
+// - *VaultError: An error object if the operation fails; otherwise, nil.
+func (v *VaultClient) WriteSecret(path, key, value string) *VaultError {
+	if err := func() *VaultError {
+		switch {
+		case v.client == nil:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorClient)}
+		case len(path) == 0:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorNoKeyPath)}
+		case len(key) == 0:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorNoKeyName)}
+		case len(value) == 0:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorNoKeyData)}
+		default:
+			return nil
+		}
+	}(); err != nil {
+		return err
+	}
```

Comment:
This is not a standard Go pattern. I don't know why we'd write the code this way. 
```suggestion
	switch {
	case v.client == nil:
		return &VaultError{Message: fmt.Sprintf(VaultErrorClient)}
	case len(path) == 0:
		return &VaultError{Message: fmt.Sprintf(VaultErrorNoKeyPath)}
	case len(key) == 0:
		return &VaultError{Message: fmt.Sprintf(VaultErrorNoKeyName)}
	case len(value) == 0:
		return &VaultError{Message: fmt.Sprintf(VaultErrorNoKeyData)}
	default:
		return nil
	}
```

Reply:
fixed on new pull request